### ISSUE TITLE
Dota Underlords trace GL_OUT_OF_MEMORY errors

### ISSUE THREAD

**Issue by danginsburg** [google/gapid#2875]
I'm trying to trace/replay of our game Dota Underlords which recently launched on Android (you can grab it free on the Play Store).  I am using gapid-1.6.0 on Windows. When I capture a trace on Google Pixel 3 XL and try to replay it I see no framebuffer images and the "Report" tab says that each draw had GL_INVALID_OPERATION and a bunch of GL_OUT_OF_MEMORY errors making the replay unusable.

Attached is a trace from the Pixel 3 XL showing the problem.

[com.valvesoftware.underlords.main_20190625_0759.zip](https://github.com/google/gapid/files/3325158/com.valvesoftware.underlords.main_20190625_0759.zip)

**Comment by pmuetschard** [comment]
Thank you for the bug report and thanks for the trace. The error messages are a bit misleading and it has nothing to do with running out of memory. We fail to properly trace and replay this application because we do not correctly implement the `EXT_buffer_storage` extension currently. Especially the coherent memory portions of the extension are going to be quite a bit of work...

Do you have a way of turning off the app's usage of that extension?

https://www.khronos.org/registry/OpenGL/extensions/EXT/EXT_buffer_storage.txt

**Comment by danginsburg** [comment]
> Do you have a way of turning off the app's usage of that extension?

Yes, I'll give that a try.

**Comment by danginsburg** [comment]
I tried again without buffer storage.  There are still a bunch of Fatal GL_INVALID_OPERATION errors.  When I click on Draws it seems to often say that the framebuffer isn't available.  In general, something seems very wrong - I'm not really able to use the tool.

Also - our engine will not use GL_EXT_buffer_storage if the extension is not exposed.  Would it make sense for gapid to filter it from the list so traces still work without app modification if it's not really supported?

Here's the trace: https://drive.google.com/file/d/1SXFW883CvGJF1yFzJkRYP_tlR2l2nUtc/view?usp=sharing

**Comment by AWoloszyn** [comment]
You are correct we should be doing this. We are already doing this in Vulkan, and we should do the same for GL. I have opened https://github.com/google/gapid/issues/2884 to track.
